rbd/snapshot_namespace_test.go

```diff
@@ -79,3 +79,78 @@ func TestGetSnapNamespaceType(t *testing.T) {
 		assert.Equal(t, nsType, SnapNamespaceTypeTrash)
 	})
 }
+
+func TestGetSnapTrashNamespace(t *testing.T) {
+	conn := radosConnect(t)
+	defer conn.Shutdown()
+
+	poolname := GetUUID()
+	err := conn.MakePool(poolname)
+	require.NoError(t, err)
+	defer conn.DeletePool(poolname)
+
+	ioctx, err := conn.OpenIOContext(poolname)
+	require.NoError(t, err)
+	defer ioctx.Destroy()
+
+	imageName := "parent"
+	snapName := "mySnap"
+	cloneName := "myClone"
+	options := NewRbdImageOptions()
+	defer options.Destroy()
+	err = options.SetUint64(ImageOptionOrder, uint64(testImageOrder))
+	assert.NoError(t, err)
+	err = options.SetUint64(ImageOptionFeatures, 1)
+	assert.NoError(t, err)
+
+	err = CreateImage(ioctx, imageName, testImageSize, options)
+	assert.NoError(t, err)
+
+	img, err := OpenImage(ioctx, imageName, NoSnapshot)
+	assert.NoError(t, err)
+	defer func() {
+		assert.NoError(t, img.Close())
+		assert.NoError(t, img.Remove())
+	}()
+
+	snapshot, err := img.CreateSnapshot(snapName)
+	assert.NoError(t, err)
+
+	optionsClone := NewRbdImageOptions()
+	defer optionsClone.Destroy()
+	err = optionsClone.SetUint64(ImageOptionCloneFormat, 2)
+	assert.NoError(t, err)
+
+	// Create a clone of the image using the same snapshot.
+	err = CloneImage(ioctx, imageName, snapName, ioctx, cloneName, optionsClone)
+	assert.NoError(t, err)
+	defer func() { assert.NoError(t, RemoveImage(ioctx, cloneName)) }()
+
+	// Check the name of the snapshot.
+	snapInfoList, err := img.GetSnapshotNames()
+	assert.NoError(t, err)
+	snapInfo := snapInfoList[0]
+	assert.Equal(t, snapInfo.Name, snapName)
+
+	// Remove the snapshot.
+	err = snapshot.Remove()
+	assert.NoError(t, err)
+
+	// Snapshot would move to the trash because linked clone is still there.
+	nsType, err := img.GetSnapNamespaceType(snapInfo.Id)
```

Comment:
can you add a test where you call it with an invalid image and id?

Reply:
Added a couple of test cases.